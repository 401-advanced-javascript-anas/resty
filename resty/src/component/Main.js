import React from 'react';
import ReactDOM from 'react-dom'
import './css/main.scss'

import './Results'

class Main extends React.Component {
    constructor (props) {
        super(props);
        this.state = {
            url: '',
            method: '',
            request: {}

        };
    }
// **********************************************
handleSubmit = async e => {
    // fetch data form API 
    e.preventDefault();
    let url = this.state.url;
    let raw = await fetch(url); // star wars API
    let data = await raw.json();
    let results = {
        Headers: raw.headers,
        Response: data
    }
    this.props.handler(results)
}



// **********************************


    handleUrl = e =>{
        let url = e.target.value;
        this.setState({url});
    }

    handleClick = e =>{
        e.preventDefault();
        let results = document.getElementById('div2')
        let words = <div>{this.state.method} <span>{this.state.url}</span> </div>
        ReactDOM.render(words, results);
    }

    handleMethod = e =>{
        let method = e.target.value;
        this.setState({method});

    }

    render() {
        return (
            <section>
                <form onSubmit={this.handleSubmit}>

            <div class="div1">
                <h2>URL:</h2>
                
   
                <input type="url" id="url" onChange={this.handleUrl} />
                <button type="submit" >GO!</button>
            </div>

            <div class="div3">
               <input type="radio" name="method" onChange={this.handleMethod} value="get"/>GET
               <input type="radio" name="method" onChange={this.handleMethod} value="post"/>POST
               <input type="radio" name="method" onChange={this.handleMethod} value="put"/>PUT
               <input type="radio" name="method" onChange={this.handleMethod} value="delete"/>DELETE
            </div>

            <div id="div2">
            </div>

            </form>
            </section>
        )
    }



}

export default Main;