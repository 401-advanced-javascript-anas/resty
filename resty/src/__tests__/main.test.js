import React from 'react';
import { shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Main from '../component/Main';


describe('React Form test',()=>{
    it('shoyld work when using the Button',()=>{
      let app = shallow(<Main />);
      expect(app.find('button').exists()).toBeTruthy();
    });
  
    it('should changes the mehod on click',()=>{
      let app = mount(<Main />);
      let input = app.find('input').at(1);
      input.simulate('click');
      console.log('******************************',input)
      expect(app.state('method')).toBe('get');
    });
    it('renders correctly', ()=> {
        const tree = renderer.create(<Main />).toJSON();
        expect(tree).toMatchSnapshot();
    });

  });