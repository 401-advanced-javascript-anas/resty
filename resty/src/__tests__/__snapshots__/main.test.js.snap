// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test renders correctly 1`] = `
<section>
  <form>
    <div
      class="div1"
    >
      <h2>
        URL:
      </h2>
      <input
        id="url"
        onChange={[Function]}
        type="url"
      />
      <button
        onClick={[Function]}
      >
        GO!
      </button>
    </div>
    <div
      class="div3"
    >
      <input
        name="method"
        onClick={[Function]}
        type="radio"
        value="get"
      />
      GET
      <input
        name="method"
        onClick={[Function]}
        type="radio"
        value="post"
      />
      POST
      <input
        name="method"
        onClick={[Function]}
        type="radio"
        value="put"
      />
      PUT
      <input
        name="method"
        onClick={[Function]}
        type="radio"
        value="delete"
      />
      DELETE
    </div>
    <div
      id="div2"
    />
  </form>
</section>
`;

exports[`React Form test renders correctly 1`] = `
<section>
  <form>
    <div
      class="div1"
    >
      <h2>
        URL:
      </h2>
      <input
        id="url"
        onChange={[Function]}
        type="url"
      />
      <button
        onClick={[Function]}
      >
        GO!
      </button>
    </div>
    <div
      class="div3"
    >
      <input
        name="method"
        onClick={[Function]}
        type="radio"
        value="get"
      />
      GET
      <input
        name="method"
        onClick={[Function]}
        type="radio"
        value="post"
      />
      POST
      <input
        name="method"
        onClick={[Function]}
        type="radio"
        value="put"
      />
      PUT
      <input
        name="method"
        onClick={[Function]}
        type="radio"
        value="delete"
      />
      DELETE
    </div>
    <div
      id="div2"
    />
  </form>
</section>
`;
